### PR TITLE
WIP: [dom-gpu] COSMO-POMPA crlim cdt/20.06

### PR DESCRIPTION
**Pull request by omlins**

### CODE REVIEW

**Comment by omlins** [comment]
NOTE: currently it builds, but fails with the segfault reported in the Cray Case. The used modules are subject to change depending on the tests. Hopefully, in the end we can have a single consistent set of modules for the build of COSMO and all recipes.

File: easybuild/easyconfigs/s/STELLA/STELLA-4a5f9c5-CrayGNU-20.06-crclim-double.eb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+toolchainopts = {'verbose': False}
+
+# c2sm repo is private (+ no official release)
+sources = [{'filename': '/apps/common/UES/easybuild/sources/s/%(name)s/%(namelower)s_%(version)s.tar.gz'}]
+
+builddependencies = [
+    ('CMake', '3.12.0', '', True),  # 3.14.5 #3.12.0
+    ('cudatoolkit/10.1.105_3.27-7.0.1.1_4.1__ga311ce7', EXTERNAL_MODULE),
+    # ('cudatoolkit/10.2.89_3.28-7.0.1.1_2.1__g88d3d59', EXTERNAL_MODULE),
+    ('gcc/8.3.0', EXTERNAL_MODULE),  # Ensure that this is loaded. Fails if not.
```
Comment (gppezzi) [inline easybuild/easyconfigs/s/STELLA/STELLA-4a5f9c5-CrayGNU-20.06-crclim-double.eb:21]: Is this still needed? it should be the default (or loaded by the CrayGNU toolchain)?

File: easybuild/easyconfigs/l/libgrib1_crclim/libgrib1_crclim-a1e4271-CrayCCE-20.06.eb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+toolchain = {'name': 'CrayCCE', 'version': '20.06'}
+
+# c2sm repo is private (+ no official release)
+sources = ['/apps/common/UES/easybuild/sources/l/%(name)s/%(name)s_%(version)s.tar.bz2']
+
+dependencies = [
+    ('cray-netcdf/4.7.3.3', EXTERNAL_MODULE),
+]
+
+prebuildopts = 'pwd; ls -l ;cd libgrib1_cosmo/source ; cat Makefile.daint.cray | perl -pe "s/-hnosecond_underscore//g" > Makefile &&'
```
[the remark this answers is not in the document]
Comment (omlins) [reply easybuild/easyconfigs/l/libgrib1_crclim/libgrib1_crclim-a1e4271-CrayCCE-20.06.eb:19]: Yes, sure; this is only for debugging. However, I would keep it until things work.

File: easybuild/easyconfigs/s/STELLA/STELLA-4a5f9c5-CrayGNU-20.06-crclim-double.eb (hunk 1 above)
Comment (omlins) [reply easybuild/easyconfigs/s/STELLA/STELLA-4a5f9c5-CrayGNU-20.06-crclim-double.eb:21]: I added this to ensure that the right gcc version is loaded. As it says in the comment: it ensures that it is loaded, else it fails. I will remove this again once everything works if you think I should or I improve the comment.

**Comment by omlins** [comment]
NOTE: at the moment this PR contains still the Dycore and COSMO recipes for CUDA 10.1 (the Stella recipe was modified to serve for CUDA 10.2).

**Comment by lucamar** [comment]
I have updated the recipe of `COSMO_pompa` and its dependencies with `cdt/20.08` and submitted the pull request #1849, therefore I'm closing this pull request (I have also merged the suggestions of this request there).

**Pull request closed without merging**